idea/src/org/jetbrains/kotlin/idea/inspections/KotlinThrowableNotThrownInspection.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2010-2019 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.idea.inspections
+
+import com.intellij.codeInspection.ProblemHighlightType
+import com.intellij.codeInspection.ProblemsHolder
+import com.intellij.psi.search.searches.ReferencesSearch
+import com.siyeh.ig.psiutils.TestUtils
+import org.jetbrains.kotlin.builtins.DefaultBuiltIns
+import org.jetbrains.kotlin.descriptors.ClassDescriptor
+import org.jetbrains.kotlin.descriptors.ConstructorDescriptor
+import org.jetbrains.kotlin.idea.caches.resolve.analyze
+import org.jetbrains.kotlin.idea.caches.resolve.resolveToCall
+import org.jetbrains.kotlin.psi.*
+import org.jetbrains.kotlin.psi.psiUtil.getParentOfTypes
+import org.jetbrains.kotlin.resolve.BindingContext
+import org.jetbrains.kotlin.resolve.descriptorUtil.isSubclassOf
+import org.jetbrains.kotlin.resolve.lazy.BodyResolveMode
+import org.jetbrains.kotlin.types.isNullable
+import org.jetbrains.kotlin.types.typeUtil.isNothing
+
+class KotlinThrowableNotThrownInspection : AbstractKotlinInspection() {
+    override fun buildVisitor(holder: ProblemsHolder, isOnTheFly: Boolean) = callExpressionVisitor(fun(callExpression) {
+        val calleeExpression = callExpression.calleeExpression ?: return
+        if (TestUtils.isInTestSourceContent(callExpression)) return
+        val resultingDescriptor = callExpression.resolveToCall()?.resultingDescriptor ?: return
```

Comment:
Consider using a heuristics to prevent resolve here: we could first check whether `Exception` or `Error` is a substring in `calleeExpression`, and if not, just return (I'm not quite sure is it a good idea or not, decide it yourself)